### DISCUSSION TITLE
how to border the middle cell

### DISCUSSION THREAD

**Discussion by radekjindra** [kryslt/KControls#62] [Q&A] [answered]
I would like to ask you for an example of how to border the middle cell in the attached table. And with different line thicknesses. Thank you.

    // var TBLM  : TKMemoTable;
    TBLM := Protokol_ZKP.Blocks.AddTable;

    TBLM.ColCount := 3;
    TBLM.ColWidths[0]:=010;
    TBLM.ColWidths[1]:=010;
    TBLM.ColWidths[2]:=010;

    TBLM.RowCount := 3;
    TBLM.RowHeights[0]:=010;
    TBLM.RowHeights[1]:=010;
    TBLM.RowHeights[2]:=010;

    TBLM.Cells[0, 0].Blocks.AddTextBlock('0');
    TBLM.Cells[0, 1].Blocks.AddTextBlock('0');
    TBLM.Cells[0, 2].Blocks.AddTextBlock('0');
    TBLM.Cells[1, 0].Blocks.AddTextBlock('0');
    TBLM.Cells[1, 1].Blocks.AddTextBlock('X');
    TBLM.Cells[1, 2].Blocks.AddTextBlock('0');
    TBLM.Cells[2, 0].Blocks.AddTextBlock('0');
    TBLM.Cells[2, 1].Blocks.AddTextBlock('0');
    TBLM.Cells[2, 2].Blocks.AddTextBlock('0');
    // attempt to delimit a cell
    TBLM.Cells[1, 1].BlockStyle.BorderWidths.Right    :=5;
    TBLM.Cells[1, 1].BlockStyle.BorderWidths.Right    :=5;
    TBLM.Cells[1, 1].BlockStyle.BorderWidths.Top      :=1;
    TBLM.Cells[1, 1].BlockStyle.BorderWidths.Bottom   :=1;

    TBLM.CellStyle.Brush.Color := clWhite;
    TBLM.ApplyDefaultCellStyle;

**Comment by kryslt** [comment, marked as answer]
Your code should be:
```
// var TBLM  : TKMemoTable;
TBLM := Protokol_ZKP.Blocks.AddTable;

TBLM.ColCount := 3;
TBLM.ColWidths[0]:=010;
TBLM.ColWidths[1]:=010;
TBLM.ColWidths[2]:=010;

TBLM.RowCount := 3;
TBLM.RowHeights[0]:=010;
TBLM.RowHeights[1]:=010;
TBLM.RowHeights[2]:=010;

TBLM.Cells[0, 0].Blocks.AddTextBlock('0');
TBLM.Cells[0, 1].Blocks.AddTextBlock('0');
TBLM.Cells[0, 2].Blocks.AddTextBlock('0');
TBLM.Cells[1, 0].Blocks.AddTextBlock('0');
TBLM.Cells[1, 1].Blocks.AddTextBlock('X');
TBLM.Cells[1, 2].Blocks.AddTextBlock('0');
TBLM.Cells[2, 0].Blocks.AddTextBlock('0');
TBLM.Cells[2, 1].Blocks.AddTextBlock('0');
TBLM.Cells[2, 2].Blocks.AddTextBlock('0');
// attempt to delimit a cell
TBLM.CellStyle.Brush.Color := clWhite;
TBLM.ApplyDefaultCellStyle;

// attempt to delimit a cell - custom border widths must be specified after ApplyDefaultCellStyle
TBLM.Cells[1, 1].RequiredBorderWidths.Left :=5;
TBLM.Cells[1, 1].RequiredBorderWidths.Right :=5;
TBLM.Cells[1, 1].RequiredBorderWidths.Top :=1;
TBLM.Cells[1, 1].RequiredBorderWidths.Bottom :=1;
```
